optuna/trial/_trial.py

```diff
@@ -378,25 +380,59 @@ def objective(trial):
                 Upper endpoint of the range of suggested values. ``high`` is included in the range.
             step:
                 A step of discretization.
+
+                .. note::
+                    Note that :math:`\\mathsf{high}` is modified if the range is not divisible by
+                    :math:`\\mathsf{step}`. Please check the warning messages to find the changed
+                    values.
+
+                .. note::
+                    The method returns one of the values in the sequence
+                    :math:`\\mathsf{low}, \\mathsf{low} + \\mathsf{step}, \\mathsf{low} + 2 *
+                    \\mathsf{step}, \\dots, \\mathsf{low} + k * \\mathsf{step} \\le
+                    \\mathsf{high}`, where :math:`k` denotes an integer.
+
+                .. note::
+                    The ``step != 1`` and ``log`` arguments cannot be used at the same time.
+                    To set the ``step`` argument :math:`\\mathsf{step} \\le 2`, set the
+                    ``log`` argument to ``False``.
             log:
                 A flag to sample the value from the log domain or not.
-                If ``log`` is true, at first, the range of suggested values is divided into grid
-                points of width ``step``. The range of suggested values is then converted to a log
-                domain, from which a value is uniformly sampled. The uniformly sampled value is
-                re-converted to the original domain and rounded to the nearest grid point that we
-                just split, and the suggested value is determined.
-                For example,
-                if `low = 2`, `high = 8` and `step = 2`,
-                then the range of suggested values is divided by ``step`` as `[2, 4, 6, 8]`
-                and lower values tend to be more sampled than higher values.
+
+                .. note::
+                    If ``log`` is true, at first, the range of suggested values is divided into
+                    grid points of width ``1``. The range of suggested values is then converted to
```

Comment:
[nit] We can remove the highlight like the other literals.

![image](https://user-images.githubusercontent.com/3255979/84347089-9cd41e00-abec-11ea-819a-adb0640569ae.png)

```suggestion
                    grid points of width 1. The range of suggested values is then converted to
```